seqr/views/utils/permissions_utils.py

```diff
@@ -37,33 +63,64 @@ def check_project_permissions(project, user, **kwargs):
         user=user, project=project))
 
 
-def check_user_created_object_permissions(obj, user):
-    if user.is_staff or obj.created_by == user:
+def check_user_created_object_permissions(obj, user, session=None):
+    if is_staff(user, session) or obj.created_by == user:
         return
     raise PermissionDenied("{user} does not have edit permissions for {object}".format(user=user, object=obj))
 
 
-def check_multi_project_permissions(obj, user):
+def check_multi_project_permissions(obj, user, session=None):
     for project in obj.projects.all():
         try:
-            check_project_permissions(project, user)
+            check_project_permissions(project, user, session=session)
             return
         except PermissionDenied:
             continue
     raise PermissionDenied("{user} does not have view permissions for {object}".format(user=user, object=obj))
 
 
-def get_projects_user_can_view(user):
+def _get_anvil_projects_user_can_view(user, session):
+    """
+    . Fetch a workspace list with a false “public” attribute
+    . If using a service account, filter out those user doesn’t have access
+    . Get a corresponding project list of the workspaces
+    . General project jsons
+    """
+    # the 'service_account_session' below will be replaced by 'session' after seqr client ID is whitelisted
+    session = service_account_session
+    is_staff_user = session.is_staff(user.anviluser.anvil_username)
+    requested_fields = 'public,workspace.name,workspace.namespace,workspace.workspaceId'
+    workspace_list = session.list_workspaces(requested_fields)
+    workspaces = []
+    for ws in workspace_list:
+        if not ws['public']:
+            if is_staff_user:
+                workspaces.append(ws['workspace']['name'])
+            else:
+                try:
+                    acl = session.get_workspace_acl(ws['workspace']['namespace'], ws['workspace']['name'])
+                except Exception:
```

Comment:
I thought it could be a `normal` case that get_workspace_acl would fail. But, actually, it fails only when there is an error. Remove the exception catching.